chalk-solve/src/wf.rs

```diff
@@ -486,29 +492,233 @@ fn compute_assoc_ty_goal<I: Interner>(
     ))
 }
 
-/// Computes a goal to prove Sized constraints on a struct definition.
-/// Struct is considered well-formed (in terms of Sized) when it either
-/// has no fields or all of it's fields except the last are proven to be Sized.  
-fn compute_struct_sized_constraint<I: Interner>(
-    db: &dyn RustIrDatabase<I>,
-    fields: &[Ty<I>],
-) -> Option<Goal<I>> {
-    if fields.len() <= 1 {
-        return None;
+struct WellKnownConstraints {}
```

Comment:
Can you add some comments to this struct and its methods below expressing what they are doing?